temporal-sdk/src/main/java/io/temporal/internal/common/WorkflowExecutionUtils.java

```diff
@@ -229,15 +234,17 @@ public static HistoryEvent getInstanceCloseEvent(
               .setWaitNewEvent(true)
               .setNextPageToken(pageToken)
               .build();
+
       long elapsed = System.currentTimeMillis() - start;
-      Deadline expiration = Deadline.after(unit.toMillis(timeout) - elapsed, TimeUnit.MILLISECONDS);
```

Comment:
What there anything wrong with using Deadline?

Reply:
The creation of Deadline was useless here. We were converting millis into Deadline only to covert Deadline immediately back to millis. Not wrong, just pointless.